src/components/card/Card.tsx

```diff
@@ -0,0 +1,86 @@
+import { useContext } from 'react';
+import { useLocation, useParams } from 'react-router-dom';
+import { GlobalContext } from '../../reducer';
+import { Product } from '../../types/product';
+import { AddToCart } from '../addToCart/AddToCard';
+import { AddToFavorite } from '../addToFavorite/AddToFavorite';
+import { NavLinkCustom } from '../navLink/NavLinkCustom';
+import './card.scss';
+
+type Props = {
+  product: Product;
+};
+
+export const Card: React.FC<Props> = ({ product }) => {
+  // eslint-disable-next-line no-empty-pattern
+  const [{}, dispatch] = useContext(GlobalContext);
```

Comment:
would be better to export object from context like `store={{ store, dispatch }}` and use only dispatch here
but anyway
```suggestion
  const [_, dispatch] = useContext(GlobalContext);
```